Install gradle archives during setup.

pedropombeiro opened the pull request:
Fixes broken `make release-android` step on new machines.

Addresses the same issue as https://github.com/status-im/status-react/pull/5110, but on the setup script.
Tested the build on a new machine where it was previously failing, and build now passes on the first attempt.

status: ready <!-- Can be ready or wip -->

Commit:
Install Android NDK and gradle archives during setup.

Fixes broken `make release-android` step on new machines.

Signed-off-by: Pedro Pombeiro <pombeirp@users.noreply.github.com>

jeluard commented:
Does this depends on the ndk installation?

pedropombeiro commented:
I believe so @jeluard. Should we show a message to the user if the NDK is not detected?

pedropombeiro commented:
I hadn't seen yet that the build failed. So I guess that the problem is that the CI server doesn't actually need this in order to do its build. Maybe we should show a warning when the NDK is not detected, but not actually fail?

jeluard commented on scripts/setup:
```diff
@@ -46,9 +46,16 @@ dependency_setup \
+    "pushd android && ./gradlew react-native-android:installArchives && popd"
+else
+    cecho "@yellow[[Android NDK installation not detected. If you plan to build for Android, \
+please follow the instructions at https://docs.status.im/docs/build_status_linux.html to install the Android NDK:]]"
```
How about if you build android on OSX?

pedropombeiro replied:
will add a check and display appropriate link

jeluard reviewed (commented):
Do you think it would be possible to have the script install the NDK ?

pedropombeiro commented:
@jeluard @rasom need a new review as the approach has changed slightly: after discussing with Dmitry N who's on OSX, I've changed approach to rely on `local.properties` file content instead of environment variables. This is more reliable and affects only status-react. If `ndk.dir` isn't defined, it will download the NDK to `~/Android/Sdk` and update the `ndk.dir` value to point to it.

pedropombeiro commented:
Added a minor improvement so that we refer the user to the correct documentation for the host platform in case the Android SDK is not present on the machine, and due to that, moved the script which was getting unwieldy to `scripts/lib/setup/installers.sh` so that `scripts/setup` stays as clean as possible.

Will merge to develop once build passes if there's nothing against this (changes are not significant relative to what was already reviewed).
